portality/static/js/formulaic.js

```diff
@@ -2045,6 +2045,64 @@ var formulaic = {
             };
 
             this.init()
-        }
+        },
+        newIssnLink : function(params) {
+            return edges.instantiate(formulaic.widgets.IssnLink, params)
+        },
+        IssnLink : function(params) {
+            this.fieldDef = params.fieldDef;
+            this.form = params.formulaic;
+            this.issn = params.issn;
+
+            this.ns = "formulaic-issnlink";
+
+            this.link = false;
+            this.url = "https://portal.issn.org/resource/ISSN/";
+
+            this.init = function() {
+                var elements = this.form.controlSelect.input(
+                    {name: this.fieldDef.name});
+                edges.on(elements, "keyup.IssnLink", this, "updateUrl");
+
+                for (var i = 0; i < elements.length; i++) {
+                    this.updateUrl(elements[i]);
+                }
+            };
+
+            this.updateUrl = function(element) {
+                var that = $(element);
+                var val = that.val();
+                var id = edges.css_id(this.ns, this.fieldDef.name);
+
+                var match = val.match(/[d0-9]{4}-{0,1}[0-9]{3}[0-9xX]{1}/);
+
+                if (val && match) {
+                    if (this.link) {
+                        this.link.text(val);
+                        this.link.attr("href", val);
+                    } else {
+                        var url = this.url + val;
+                        var classes = edges.css_classes(this.ns, "visit");
+                        that.after('<p><small><a id="' + id + '" class="' + classes + '" rel="noopener noreferrer" target="_blank" href="' + url + '">' + url + '</a></small></p>');
+
+                        var selector = edges.css_id_selector(this.ns, this.fieldDef.name);
+                        this.link = $(selector, this.form.context);
+                        var client = new XMLHttpRequest();
+                        client.open("GET", "https://portal.issn.org/resource/ISSN/0000-0000", true);
+                        client.send();
+                        client.onreadystatechange = function() {
+                            if(this.readyState == this.HEADERS_RECEIVED) {
+                                console.log(client.getResponseHeader("Location"));
+                            }
+                        }
```

Comment:
What is this bit doing?  My feeling is that we shouldn't be contacting the portal directly in the javascript, and it feels like this might just be for development testing?  Can we take it out before we finish this.

Reply:
Yep, it should be cleaned up. I tried to find a way to find out whether the link results in redirection to indicate whether the ISSN can be found on the portal.issn.org (if it can't, the link redirects to the homepage)